packages/dds/tree/src/simple-tree/unhydratedFlexTreeFromInsertable.ts

```diff
@@ -406,6 +410,44 @@ function objectToFlexContent(
 	return [{ type: brand(schema.identifier) }, fields];
 }
 
+/**
+ * Transforms data under an Object schema.
+ * @param data - The tree data to be transformed. Must be a Record-like object.
+ * @param schema - The schema associated with the value.
+ */
+function recordToFlexContent(data: FactoryContent, schema: TreeNodeSchema): FlexContent {
+	assert(isRecordNodeSchema(schema), "Expected a Record schema.");
+	if (!(typeof data === "object" && data !== null)) {
+		throw new UsageError(`Input data is incompatible with Record schema: ${data}`);
+	}
+
+	const allowedChildTypes = normalizeAllowedTypes(schema.info as ImplicitAllowedTypes);
+
+	const fieldsIterator: Iterable<readonly [string, InsertableContent]> = Object.entries(data);
```

Comment:
Once you have the `Iterable<readonly [string, InsertableContent]>` the rest of this function is the same as `mapToFlexContent`

Rather than duplicating this code, I think it should be shared.